### PR TITLE
[MRG+2] For #205 Add LogEndogTransformer

### PR DESCRIPTION
**Pull request by christopher-siewert**
# Description

I have added a preprocessing `LogEndogTransformer` class that extends the existing `BoxCox` class. I wrote two simple tests and added the log class to the docs.

This is WIP because the functionality is a bit strange. I tried to describe the problem in #205, but I think a pull request will be clearest. 

`LogEndogTransformer` inherited the `fit` and `fit_transform` methods. These don't make sense for a simple stateless transform like the log. The fact that you have to "fit" the log transform before you use it is strange. 

Take a look and let me know what you think.

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] Documentation change

# How Has This Been Tested?

- [X] make test

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pmdarima/preprocessing/endog/log.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    floor : float, optional (default=1e-16)
+        A positive value that truncate values to if there are values in ``y``
+        that are zero or negative and ``neg_action`` is not 'raise'. Note that
+        if values are truncated, invertibility will not be preserved, and the
+        transformed array may not be perfectly inverse-transformed.
+    """
+    def __init__(self, lmbda=0, neg_action="raise", floor=1e-16):
+
+        super().__init__(0, lmbda2=lmbda, neg_action=neg_action, floor=floor)
```
Comment (tgsmith61591) [inline pmdarima/preprocessing/endog/log.py]: I think this is a really clever way to handle it (`lmbda2=lmbda`)

**Comment by aaronreidsmith** [comment]
@christopher-siewert Do you want to go ahead and add yourself to `AUTHORS.md` before we merge this?

**Comment by aaronreidsmith** [comment]
Might be good to update `doc/whats_new.rst` under v.1.5.0 as well!

**Comment by christopher-siewert** [comment]
I added a `fit` method to `LogEndogTransformer` to make the documentation clearer (Otherwise it uses the boxcox `fit` method documentation which has different parameter names and was confusing).
I added my name to `AUTHORS.md`.
I added the change to `doc/whats_new.rst`

**Pull request merged**